runtime/dynlink.c

```diff
@@ -173,35 +176,28 @@ void caml_build_primitive_table(char_os * lib_path,
      - directories specified in the CAML_LD_LIBRARY_PATH
      - directories specified in the executable
      - directories specified in the file <stdlib>/ld.conf */
-  tofree1 = caml_decompose_path(&caml_shared_libs_path,
-                                caml_secure_getenv(T("CAML_LD_LIBRARY_PATH")));
+  caml_decompose_path(&caml_shared_libs_path,
+                      caml_secure_getenv(T("CAML_LD_LIBRARY_PATH")));
   if (lib_path != NULL)
     for (p = lib_path; *p != 0; p += strlen_os(p) + 1)
       caml_ext_table_add(&caml_shared_libs_path, p);
-  tofree2 = caml_parse_ld_conf();
+  caml_parse_ld_conf();
```

Comment:
I think it might be worth an explicit comment noting that the buffers mustn't be freed?